lib/ScriptFileStorage.js

```diff
@@ -224,12 +233,16 @@ $class.findAllApplicationScripts = function(startDirectory, mainScriptFile, call
     function(err, results) {
       if (err) return callback(err);
 
+      debug('findAllApplicationScripts filtering start');
+
       var files = results[0].concat(results[1]);
       // filter out duplicates and files to hide
       files = files.filter(function(elem, ix, arr) {
         return arr.indexOf(elem) >= ix && !this._scriptManager.isScriptHidden(elem);
```

Comment:
I suspect that `arr.indexOf` can be the root cause of low performance, as this is an `O(n^2)` algorithm.